### PR TITLE
Fixes problems caused by stateful PartitionAwareOperationFactory

### PR DESCRIPTION
**Pull request by ahmetmircik**
Thanks @Donnerbart for finding this problem

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/map/impl/operation/PartitionWideEntryWithPredicateOperationFactory.java [hunk 1]
```diff
@@ -43,38 +44,74 @@
...
 
     private String name;
     private EntryProcessor entryProcessor;
     private Predicate predicate;
 
-    private transient Set<Data> keySet;
-    private transient NodeEngine nodeEngine;
-    private transient boolean hasIndex;
-    private transient Map<Integer, List<Data>> partitionIdToKeys;
+    private Map<Integer, List<Data>> partitionIdToKeysMap;
```
Comment (Donnerbart) [inline hazelcast/src/main/java/com/hazelcast/map/impl/operation/PartitionWideEntryWithPredicateOperationFactory.java]: Should this field be `transient` again, to make clear that it's a local state?

Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/map/impl/operation/PartitionWideEntryWithPredicateOperationFactory.java]: done

**Comment by Donnerbart** [comment]
:+1: (once green)

**Comment by ahmetmircik** [comment]
run-lab-run

**Comment by jerrinot** [comment]
:+1:

**Pull request merged**